helix-core/src/test/java/org/apache/helix/integration/task/TestTaskSchedulingTwoCurrentStates.java

```diff
@@ -65,7 +60,7 @@
  * sent when there are two CurrentStates.
  */
 public class TestTaskSchedulingTwoCurrentStates extends TaskTestBase {
-  private static final String DATABASE = WorkflowGenerator.DEFAULT_TGT_DB;
```

Comment:
Is it possible the old way of using WorkflowGenerator.DEFAULT_TGT_DB would have conflict, if multiple test using the same "WorkflowGenerator.DEFAULT_TGT_DB"?

Reply:
The main reason is that in before class method, "TestDB" is created as Full-Auto and in this test we are trying to change it to Semi-Auto which causes race condition between controller and test updating IdealState. So if want to have Semi-Auto resource, we need to create it separately (with different name) and leverage it in this test.

Reply:
I think the better way to fix it is not call super.beforeClass() in line 72/77. This way, we won't create the cluster/full auto etc and participant that we need to stop immediately. 

Instead, just creates cluster, controller and participant in this before test. What is your take?

Reply:
Your point is somehow valid. But I still think this is better option because this way we are saving many duplicated code which will be hard to maintain and most of the code is written in the parent class. Instead of that, as what I proposed in this PR, we can separate the DATABASE name and create new one which includes this class name and will be unique. We have same behavior for most of the Task Framework related test and workflow names have been derived from class name. This way the tests is easier to maintain and more understandable.